yt_dlp/extractor/drtuber.py

```diff
@@ -21,84 +23,201 @@ class DrTuberIE(InfoExtractor):
                 formats.append({
                     'format_id': format_id,
-                    'quality': 2 if format_id == 'hq' else 1,
+                    'quality': qualities.get(format_id) or format_id,
+                    'height': int_or_none(qualities.get(format_id)[:-1]),
                     'url': video_url,
                 })
+        self._check_formats(formats, video_id)
 
         duration = int_or_none(video_data.get('duration')) or parse_duration(
             video_data.get('duration_format'))
 
-        title = self._html_search_regex(
-            (r'<h1[^>]+class=["\']title[^>]+>([^<]+)',
+        title = video_data.get('title') or self._html_search_regex(
+            (r'<div.*class=[\'"]caption[\'"].*?><h2>(.+?)</h2>',
+             r'<h1[^>]+class=["\']title[^>]+>([^<]+)',
              r'<title>([^<]+)\s*@\s+DrTuber',
              r'class="title_watch"[^>]*><(?:p|h\d+)[^>]*>([^<]+)<',
              r'<p[^>]+class="title_substrate">([^<]+)</p>',
              r'<title>([^<]+) - \d+'),
             webpage, 'title')
 
-        thumbnail = self._html_search_regex(
-            r'poster="([^"]+)"',
-            webpage, 'thumbnail', fatal=False)
+        mobile_webpage = None
+        if not title:
+            mobile_webpage = self._download_webpage(
+                f'http://m.{domain}.com/video/{video_id}',
+                video_id, 'Downloading mobile video page', fatal=False) or ''
+
+            title = strip_or_none(video_data.get('title') or self._html_search_regex(
+                (r'''<span\s[^>]*?\btitle\s*=\s*(?P<q>"|'|\b)(?P<title>[^"]+)(?P=q)\s*>''',
+                    r'''<div\s[^>]*?\bclass\s*=\s*(?P<q>"|'|\b)thumb-holder video(?P=q)>\s*<h5\b[^>]*>(?P<title>[^<]+)</h5''',
+                    r'''<span\s[^>]*?\bclass\s*=\s*(?P<q>"|'|\b)title_thumb(?P=q)>(?P<title>[^<]+)</span'''),
+                mobile_webpage, 'title', group='title'))
+
+        thumbnails = []
+        if not mobile_webpage:
+            mobile_webpage = self._download_webpage(
+                f'http://m.{domain}.com/video/{video_id}',
+                video_id, 'Downloading mobile video page', fatal=False) or ''
+
+            thumbnails = [
+                {'url': thumb_url} for thumb_url in re.findall(
+                    r'<div\s+class\s*=\s*"video-tmb-wrap"\s*>\s*<img\s+src\s*=\s*"([^"]+)"\s*/>', mobile_webpage)
+                if url_or_none(thumb_url)]
+
+        if url_or_none(video_data.get('poster')):
+            thumbnails.append({'url': video_data['poster'], 'preference': 1})
 
         def extract_count(id_, name, default=NO_DEFAULT):
             return str_to_int(self._html_search_regex(
-                rf'<span[^>]+(?:class|id)="{id_}"[^>]*>([\d,\.]+)</span>',
-                webpage, f'{name} count', default=default, fatal=False))
+                rf'<span[^>]+(?:class|id)="{id_}"[^>]*>(?P<{name}>[\d,\.]+)</span>',
+                webpage, f'{name} count', default=default, fatal=False, group=name))
```

Comment:
I don't understand why this change is necessary if all other groups are non-capturing